internal/pkg/telemetry/ddtracer.go

```diff
@@ -0,0 +1,101 @@
+package telemetry
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/xerrors"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	attrDDError           = attribute.Key("error")
+	attrDDErrorType       = attribute.Key("error.type")
+	attrDDErrorDetails    = attribute.Key("error.details")
+	attrDDErrorStackTrace = attribute.Key("error.stack")
+	stackTraceLength      = 20
+)
+
+type wrappedDDTracer struct {
+	tracer trace.Tracer
+}
+
+type wrappedDDSpan struct {
+	trace.Span
+}
+
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+func (s *wrappedDDSpan) RecordError(err error, _ ...trace.EventOption) {
+	errMsg := err.Error()
+	errType := typeStr(err)
+	s.Span.SetAttributes(
+		semconv.ExceptionMessage(errMsg),
+		semconv.ExceptionType(errType),
+		attrDDError.String(errMsg),
+		attrDDErrorType.String(errType),
+	)
+
+	// Add details
+	switch err.(type) { //nolint: errorlint
```

Comment:
`//nolint: errorlint` chyby v Go maju hierarchiu, napr. `multierror` co som uz niekde vysvetloval.
Ak error implementuje `Unwrap() error`, tak je mozne ziskat nadradenu chybu.

To sa robi cez `errors.As(err, &target)`. Tj. ak je v hierarchii nikde chyba `target` typu, tak sa ziska.

Linteru sa tu nepaci, ze nepouzivam `errors.As`, ale ja to robim schvalne, lebo mna zaujima ci ta root error ma dany interface alebo nie. To, ze ho ma nejaka vnorena chyba mi nepomoze, lebo by som ziskal iba ciastocnu informaciu.